### ISSUE TITLE
Is it possible to parse a Haskell-like range notation, e.g. [1, 3 .. 9], where the step is given by 3 - 1?

### ISSUE THREAD

**Issue by davips** [markusressel/py-range-parse#2]
The Haskell approach is similar to yours (except by different kinds of limits you offer) but has this small feature of defining the step "inductively".

**Comment by markusressel** [comment]
No, this is currently not possible. "Gaps" in general would have to be modelled by the dev manually.
Im not a math guy, but this seems like a request that is rather complex to implement. If you would like to do that feel free to give it a shot.
